hs-bindgen/test/hs-bindgen/Test/HsBindgen/Golden.hs

```diff
@@ -245,6 +245,17 @@ testCases = [
                Nothing
         , testRustBindgen = RustBindgenFail
         }
+    , (defaultTest "fun_attributes_conflict-17") {
+          testClangVersion = Just (< (18, 0, 0))
+        }
+    , (defaultTest "fun_attributes_conflict") {
```

Comment:
I hadn't realized `testClangVersion` could be used to have different tests for different versions, clever :)

Reply:
There is a downside to this approach and that is that if you want to regenerate the golden files, you'll have to run the golden tests twice with different clang versions: once with <18, once with >=18

Reply:
Actually, now that https://github.com/well-typed/hs-bindgen/pull/921#discussion_r2222744223 is done, we don't need two versions of the same golden test anymore!